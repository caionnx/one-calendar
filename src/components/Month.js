import React, { Component } from 'react';
import calendarMatrix from 'calendar-matrix';
import isToday from 'date-fns/is_today';
import WeekHead from './WeekHead';
import Day from './Day';

class Month extends Component {
  getMatrixOfDays (year, month) {
    return calendarMatrix(year, month)
  }

  isToday (year, month, day) {
    const dirtyFormat = `${year}-${month + 1}-${day}`;

    return isToday(dirtyFormat);
  }

  isWeekend (dayIndex) {
    return dayIndex === 0 || dayIndex === 6;
  }

  render () {
    const {
      name,
      year,
      index,
    } = this.props;

    return (
      <table className='Month'>
        <thead>
          <tr>
            <th colSpan='7' className='Month__Title'>{name}</th>
          </tr>
          <WeekHead />
        </thead>
        <tbody>
          {
            // Get a matrix for the month, each line is a week
            this.getMatrixOfDays(year, index).map(row => 
              <tr>
                {
                  // Now can show the day
                  row.map((day, i) => 
                    <Day
                      numeric={day}
                      isToday={this.isToday(year, index, day)}
                      isWeekend={this.isWeekend(i)} />
                  )
                }
              </tr>
            )
          }
        </tbody>
      </table>
    )
  }
}

export default Month;