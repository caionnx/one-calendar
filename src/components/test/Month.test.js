import React from 'react';
import { shallow } from 'enzyme';

import Month from '../Month';

it('render month correctly', () => {
  const wrapper = shallow(<Month name='January' index='0' year={2018} />);

  expect(wrapper).toMatchSnapshot();
});