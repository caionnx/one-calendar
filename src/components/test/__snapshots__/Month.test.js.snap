// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render month correctly 1`] = `
<table
  className="Month"
>
  <thead>
    <tr>
      <th
        className="Month__Title"
        colSpan="7"
      >
        January
      </th>
    </tr>
    <WeekHead />
  </thead>
  <tbody>
    <tr
      key="2018,0,0"
    >
      <Connect(Day)
        dayIndex={0}
        key="2018,0,-31"
        month="0"
        numeric={-31}
        year={2018}
      />
      <Connect(Day)
        dayIndex={1}
        key="2018,0,1"
        month="0"
        numeric={1}
        year={2018}
      />
      <Connect(Day)
        dayIndex={2}
        key="2018,0,2"
        month="0"
        numeric={2}
        year={2018}
      />
      <Connect(Day)
        dayIndex={3}
        key="2018,0,3"
        month="0"
        numeric={3}
        year={2018}
      />
      <Connect(Day)
        dayIndex={4}
        key="2018,0,4"
        month="0"
        numeric={4}
        year={2018}
      />
      <Connect(Day)
        dayIndex={5}
        key="2018,0,5"
        month="0"
        numeric={5}
        year={2018}
      />
      <Connect(Day)
        dayIndex={6}
        key="2018,0,6"
        month="0"
        numeric={6}
        year={2018}
      />
    </tr>
    <tr
      key="2018,0,1"
    >
      <Connect(Day)
        dayIndex={0}
        key="2018,0,7"
        month="0"
        numeric={7}
        year={2018}
      />
      <Connect(Day)
        dayIndex={1}
        key="2018,0,8"
        month="0"
        numeric={8}
        year={2018}
      />
      <Connect(Day)
        dayIndex={2}
        key="2018,0,9"
        month="0"
        numeric={9}
        year={2018}
      />
      <Connect(Day)
        dayIndex={3}
        key="2018,0,10"
        month="0"
        numeric={10}
        year={2018}
      />
      <Connect(Day)
        dayIndex={4}
        key="2018,0,11"
        month="0"
        numeric={11}
        year={2018}
      />
      <Connect(Day)
        dayIndex={5}
        key="2018,0,12"
        month="0"
        numeric={12}
        year={2018}
      />
      <Connect(Day)
        dayIndex={6}
        key="2018,0,13"
        month="0"
        numeric={13}
        year={2018}
      />
    </tr>
    <tr
      key="2018,0,2"
    >
      <Connect(Day)
        dayIndex={0}
        key="2018,0,14"
        month="0"
        numeric={14}
        year={2018}
      />
      <Connect(Day)
        dayIndex={1}
        key="2018,0,15"
        month="0"
        numeric={15}
        year={2018}
      />
      <Connect(Day)
        dayIndex={2}
        key="2018,0,16"
        month="0"
        numeric={16}
        year={2018}
      />
      <Connect(Day)
        dayIndex={3}
        key="2018,0,17"
        month="0"
        numeric={17}
        year={2018}
      />
      <Connect(Day)
        dayIndex={4}
        key="2018,0,18"
        month="0"
        numeric={18}
        year={2018}
      />
      <Connect(Day)
        dayIndex={5}
        key="2018,0,19"
        month="0"
        numeric={19}
        year={2018}
      />
      <Connect(Day)
        dayIndex={6}
        key="2018,0,20"
        month="0"
        numeric={20}
        year={2018}
      />
    </tr>
    <tr
      key="2018,0,3"
    >
      <Connect(Day)
        dayIndex={0}
        key="2018,0,21"
        month="0"
        numeric={21}
        year={2018}
      />
      <Connect(Day)
        dayIndex={1}
        key="2018,0,22"
        month="0"
        numeric={22}
        year={2018}
      />
      <Connect(Day)
        dayIndex={2}
        key="2018,0,23"
        month="0"
        numeric={23}
        year={2018}
      />
      <Connect(Day)
        dayIndex={3}
        key="2018,0,24"
        month="0"
        numeric={24}
        year={2018}
      />
      <Connect(Day)
        dayIndex={4}
        key="2018,0,25"
        month="0"
        numeric={25}
        year={2018}
      />
      <Connect(Day)
        dayIndex={5}
        key="2018,0,26"
        month="0"
        numeric={26}
        year={2018}
      />
      <Connect(Day)
        dayIndex={6}
        key="2018,0,27"
        month="0"
        numeric={27}
        year={2018}
      />
    </tr>
    <tr
      key="2018,0,4"
    >
      <Connect(Day)
        dayIndex={0}
        key="2018,0,28"
        month="0"
        numeric={28}
        year={2018}
      />
      <Connect(Day)
        dayIndex={1}
        key="2018,0,29"
        month="0"
        numeric={29}
        year={2018}
      />
      <Connect(Day)
        dayIndex={2}
        key="2018,0,30"
        month="0"
        numeric={30}
        year={2018}
      />
      <Connect(Day)
        dayIndex={3}
        key="2018,0,31"
        month="0"
        numeric={31}
        year={2018}
      />
      <Connect(Day)
        dayIndex={4}
        key="2018,0,-1"
        month="0"
        numeric={-1}
        year={2018}
      />
      <Connect(Day)
        dayIndex={5}
        key="2018,0,-2"
        month="0"
        numeric={-2}
        year={2018}
      />
      <Connect(Day)
        dayIndex={6}
        key="2018,0,-3"
        month="0"
        numeric={-3}
        year={2018}
      />
    </tr>
  </tbody>
</table>
`;
